packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.ts

```diff
@@ -877,6 +884,12 @@ const findNextHtmlTag = (text: string, startIndex: number): HtmlTag | undefined
     // No close tag
     return undefined;
   }
+
+  const textBeforeTagStart = text.substring(0, tagStartIndex + 1);
+  const skippedAngleBrackets = textBeforeTagStart?.match(/&lt;/g);
```

Comment:
We might need to keep a list of user-updated indices of angle brackets to handle this properly.